src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryMultiSink.java

```diff
@@ -131,10 +133,14 @@ protected void configureOutputSchemas(BatchSinkContext context,
             config.isTruncateTableSet(), config.getDataset(), collector);
 
         }
-
-        String outputName = String.format("%s-%s", config.getReferenceName(), tableName);
+        String fqn = BigQueryUtil.getFQN(config.getDatasetProject(),
+                                         config.getDataset(), config.getTable());
+        String referenceName = Strings.isNullOrEmpty(config.getReferenceName())
+          ? ReferenceNames.normalizeFqn(fqn)
+          : config.getReferenceName();
```

Comment:
looks like this logic is duplicated in a bunch of places. It would be better to have it in just a single place. I think it's better to have getFQN() and getReferenceName() in the config class rather than in the source/sink.